src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -6422,4 +6528,276 @@ public static class D
+
+        var interceptors = CSharpTestSource.Parse($$"""
+            using System;
+            using System.Runtime.CompilerServices;
+
+            static class Interceptors
+            {
+                [InterceptsLocation({{locationSpecifier.Version}}, "{{locationSpecifier.Data}}")]
+                public static void M1() => Console.Write(1);
+            }
+            """, "Interceptors.cs", RegularWithInterceptors);
+
+        var comp1 = CreateCompilation([interceptors, CSharpTestSource.Parse(s_attributesSource.text, s_attributesSource.path, RegularWithInterceptors)]);
+        comp1.VerifyEmitDiagnostics(
+            // Interceptors.cs(6,28): error CS9232: Cannot intercept a call in file 'Program.cs' because a matching file was not found in the compilation.
+            //     [InterceptsLocation(1, "{locationSpecifier.Data}")]
+            Diagnostic(ErrorCode.ERR_InterceptsLocationFileNotFound, $@"""{locationSpecifier.Data}""").WithArguments("Program.cs").WithLocation(6, 28));
+    }
+
+    [Fact]
+    public void Checksum_08()
+    {
+        // Duplicate file
+
+        var source = """
+            class C
+            {
+                static void M() => throw null!;
+
+                static void Main()
+                {
+                    M();
+                }
+            }
+            """;
+        var sourceTree1 = CSharpTestSource.Parse(source, path: "Program1.cs", options: RegularWithInterceptors);
+
+        var comp = CreateCompilation(sourceTree1);
+        var model = comp.GetSemanticModel(sourceTree1);
+        var node = sourceTree1.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
+        var locationSpecifier = model.GetInterceptableLocation(node)!;
+
+        var interceptors = CSharpTestSource.Parse($$"""
+            using System;
+            using System.Runtime.CompilerServices;
+
+            static class Interceptors
+            {
+                [InterceptsLocation({{locationSpecifier.Version}}, "{{locationSpecifier.Data}}")]
+                public static void M1() => Console.Write(1);
+            }
+            """, "Interceptors.cs", RegularWithInterceptors);
+
+        var comp1 = CreateCompilation([
+            sourceTree1,
+            CSharpTestSource.Parse(source, path: "Program2.cs", options: RegularWithInterceptors),
+            interceptors,
+            CSharpTestSource.Parse(s_attributesSource.text, s_attributesSource.path, RegularWithInterceptors)]);
+        comp1.GetDiagnostics().Where(d => d.Location.SourceTree == interceptors).Verify(
+            // Interceptors.cs(6,28): error CS9231: Cannot intercept a call in file 'Program1.cs' because it is duplicated elsewhere in the compilation.
+            //     [InterceptsLocation(1, "{data}")]
+            Diagnostic(ErrorCode.ERR_InterceptsLocationDuplicateFile, $@"""{locationSpecifier.Data}""").WithArguments("Program1.cs").WithLocation(6, 28));
+    }
```

Comment:
Would it be possible to update (or duplicate) the existing tests to shift our coverage to the new format? We'll need to do that at some point (when deprecating the old formats). I think starting that effort would be beneficial given that we've reached the final format (hopefully).

Reply:
For this PR I'd be happy to port/adjust as many tests as needed to test the newly added code paths. This PR is only about how to identify the intercepted call in syntax. Changes to tests which are about lowering the intercepted calls I would prefer to keep to a minimum. That work will have to happen eventually but it's a lot of churn/manual labor, especially if we have reason to think the signature of the attributes, etc., is subject to change.